### PR TITLE
Use the upstream TCL parser

### PR DESCRIPTION
**Pull request by techee**
The TCL parser is split into three parts:

tcl - for parsing pure tcl code
tcloo - for parsing tcloo OO extension of tcl
itcl - for parsing itcl OO extension of tcl

TCLOO and ITCL parsers run the TCL parser as a subparser but it isn't
possible to combine all three parsers at the same time. Since TCLOO
is the "modern" (but 10 years old) OO extension, it seems to be a better
choice for Geany (if needed, a new filetype could be introduced for
ITCL).

Since it's run as a subparser, we need to introduce also a subparser
tag type mapping for TCL (types are mapped to the same values here
because they don't conflict with the TCLOO types).

Also, the new parser reports full scope information so take care of that.
In addition, scope by the TCL parser is prefixed by :: which we need
to strip because the TM doesn't expect scope in this format.

Update the unit test based on several uctags unit tests to cover the
tags we generate with this parser.

### CODE REVIEW

File: tests/ctags/simple.tcl [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+    }
+}
 
-itcl::class itcl_class
+class create ShouldNotBeCaptured {
+}
 
-public method method1
-protected method method2
-private method method3
```
Comment (kugel-) [inline tests/ctags/simple.tcl:10]: Not sure if it matters but the new tests don't include public/private methods, TclOO supports these according to https://wiki.tcl-lang.org/page/TclOO+Tutorial, but I don't know if it ctags handles it at all?

**Review by kugel- — APPROVED** [review]
I didn't dive into the parser, rest looks good to me.

**Comment by techee** [comment]
> I didn't dive into the parser, rest looks good to me.

Thanks for having a look at this.

**Comment by elextr** [comment]
Tests pass, symbols show, LGTM.

**Pull request merged**